go.mod

```diff
@@ -1,19 +1,67 @@
 module github.com/aws-controllers-k8s/runtime
 
-go 1.14
+go 1.17
 
 require (
 	github.com/aws/aws-sdk-go v1.37.10
-	github.com/go-logr/logr v0.1.0
-	github.com/google/go-cmp v0.3.1
+	github.com/go-logr/logr v1.2.0
+	github.com/google/go-cmp v0.5.5
 	github.com/jaypipes/envutil v1.0.0
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.1.0
+	github.com/prometheus/client_golang v1.11.0
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.5.1
-	go.uber.org/zap v1.10.0
-	k8s.io/api v0.18.2
-	k8s.io/apimachinery v0.18.6
-	k8s.io/client-go v0.18.2
-	sigs.k8s.io/controller-runtime v0.6.0
+	github.com/stretchr/testify v1.7.0
+	go.uber.org/zap v1.19.1
+	k8s.io/api v0.23.0
+	k8s.io/apimachinery v0.23.0
+	k8s.io/client-go v0.23.0
+	sigs.k8s.io/controller-runtime v0.11.0
```

Comment:
I'd love to see what else we unlocked in each of these version updates. Especilaly `logr` going from `0.1.0` to `1.2.0` - I imagine there are some interesting features we might use.